core/block_dragger.js

```diff
@@ -186,6 +188,16 @@ Blockly.BlockDragger.prototype.startBlockDrag = function(currentDragDeltaXY,
   }
 };
 
+/**
+ * Fire a ui event at the start of a block drag.
+ * @private
+ */
+Blockly.BlockDragger.prototype.fireDragStartEvent_ = function() {
+  var event = new Blockly.Events.Ui(this.draggingBlock_, 'dragStart',
+      null, this.draggingBlock_.getDescendants(false));
```

Comment:
Couple of questions: 
- I'm curious why we're passing newValue here with the list of blocks, whereas oldValue on the end event below. Looking at other examples of UI events we generally only pass values to old / new if some value has changed.
- I'm also curious why we're passing this.draggingBlock_.getDescendants(false) at all? We don't seem to do that in any other events.

Also, could we perhaps do this with the one event name? ``drag`` and pass ``false,true`` as we start the drag, and ``true, false`` as we end the event.

Reply:
It's passing the list of blocks that are part of the drag. If we want to use this event for highlighting or locking blocks that are currently being dragged we'll want the full list of ids.

It's the new value when the drag starts, because those are the blocks being dragged, and the old value at the end because they are no longer being dragged. We could do this as a single drag event and just update the list of ids that are being dragged instead of different start and end events, though if we ever re-implement multi-drag that would be a problem.

Reply:
That makes sense. Thanks @RoboErikG